From 0855b6bbc1ef93ed4cdc601796e4b05b6f8f3ae1 Mon Sep 17 00:00:00 2001
From: Li Zhou <li.zhou@windriver.com>
Date: Mon, 17 Apr 2017 17:49:56 +0800
Subject: [PATCH] openl2tp: remove the bashisms from openl2tpd

Upstream-Status: Inappropriate [Only for yocto]

Signed-off-by: Li Zhou <li.zhou@windriver.com>
---
 etc/rc.d/init.d/openl2tpd | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/etc/rc.d/init.d/openl2tpd b/etc/rc.d/init.d/openl2tpd
index 1f8a92f..088155f 100755
--- a/etc/rc.d/init.d/openl2tpd
+++ b/etc/rc.d/init.d/openl2tpd
@@ -1,4 +1,4 @@
-#! /bin/bash
+#! /bin/sh
 #
 # openl2tpd          Start/Stop the OpenL2TP protocol daemon.
 #
@@ -20,10 +20,10 @@
 prog="openl2tpd"
 
 start() {
-	echo -n $"Starting $prog: "	
+	echo -n "Starting $prog: "
         if [ -e /var/lock/subsys/openl2tpd ]; then
 	    if [ -e /var/run/openl2tpd.pid ] && [ -e /proc/`cat /var/run/openl2tpd.pid` ]; then
-		echo -n $"cannot start openl2tpd: openl2tpd is already running.";
+		echo -n "cannot start openl2tpd: openl2tpd is already running.";
 		#failure $"cannot start openl2tpd: openl2tpd already running.";
 		echo
 		return 1
@@ -40,7 +40,7 @@ start() {
 	    touch /var/lock/subsys/openl2tpd
 	    if [ -n "$OPENL2TPD_CONFIG_FILE" ]; then
 		sleep 1
-		echo $"Restoring saved openl2tpd configuration..."
+		echo "Restoring saved openl2tpd configuration..."
 		/usr/bin/l2tpconfig config restore file=$OPENL2TPD_CONFIG_FILE
 		RETVAL=$?
 	    fi
@@ -49,9 +49,9 @@ start() {
 }
 
 stop() {
-	echo -n $"Stopping $prog: "
+	echo -n "Stopping $prog: "
         if [ ! -e /var/lock/subsys/openl2tpd ]; then
-	    echo -n $"cannot stop openl2tpd: openl2tpd is not running."
+	    echo -n "cannot stop openl2tpd: openl2tpd is not running."
 	    #failure $"cannot stop openl2tpd: openl2tpd is not running."
 	    echo
 	    return 1;
@@ -74,7 +74,7 @@ restart() {
 }	
 
 reload() {
-	echo -n $"Reloading openl2tp daemon configuration: "
+	echo -n "Reloading openl2tp daemon configuration: "
 	stop
 	start
 	RETVAL=$?
@@ -102,6 +102,6 @@ case "$1" in
   	[ -f /var/lock/subsys/openl2tpd ] && restart || :
 	;;
   *)
-	echo $"Usage: $0 {start|stop|status|reload|restart|condrestart}"
+	echo "Usage: $0 {start|stop|status|reload|restart|condrestart}"
 	exit 1
 esac
-- 
1.9.1

