diff --git a/samples/init.d-spmd.in b/samples/init.d-spmd.in
index d38b825..624f22d 100644
--- a/samples/init.d-spmd.in
+++ b/samples/init.d-spmd.in
@@ -1,5 +1,8 @@
 #! /bin/sh
 
+# Source function library.
+. /etc/init.d/functions
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 
@@ -27,6 +30,10 @@ case "$1" in
 	$DAEMON -k
 	echo "."
 	;;
+  status)
+	status $DAEMON;
+	exit $?
+	;;
   restart|force-reload)
   	$0 stop
 	$0 start
@@ -34,7 +41,7 @@ case "$1" in
   *)
 	N=/etc/init.d/$NAME
 	# echo "Usage: $N {start|stop|restart|reload|force-reload}" >&2
-	echo "Usage: $N {start|stop|restart|force-reload}" >&2
+	echo "Usage: $N {start|stop|status|restart|force-reload}" >&2
 	exit 1
 	;;
 esac
