From 8b69023b6dceb62da2daf67626a54c8a257fc210 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Mon, 17 Feb 2014 03:07:34 -0500
Subject: [PATCH 6/6] fix real path for dosfsck, mkdosfs

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/fstools.fc |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/fstools.fc b/policy/modules/system/fstools.fc
index a724776..a1d94ff 100644
--- a/policy/modules/system/fstools.fc
+++ b/policy/modules/system/fstools.fc
@@ -5,6 +5,7 @@
 /sbin/blockdev\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/cfdisk		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/dosfsck		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/dosfsck	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/dump		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/dumpe2fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/e2fsck		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -23,6 +24,7 @@
 /sbin/lsraid		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/make_reiser4	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/mkdosfs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/mkdosfs	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/mke2fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/mke4fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/mkfs.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
-- 
1.7.9.5

