From dbaed0d36a23c3b3fe1239f5b74381ff1eb471fb Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 17 Sep 2018 08:34:05 +0800
Subject: [PATCH] refpolicy: fix path for fstools

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/fstools.fc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/policy/modules/system/fstools.fc b/policy/modules/system/fstools.fc
index d719e22..e21d959 100644
--- a/policy/modules/system/fstools.fc
+++ b/policy/modules/system/fstools.fc
@@ -60,6 +60,7 @@
 /usr/sbin/blkid			--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/blkid\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/blockdev		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/blockdev\.util-linux		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/cfdisk		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/clubufflush		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/delpart		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -79,6 +80,7 @@
 /usr/sbin/gdisk			--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/hdparm		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/hdparm\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/hdparm\.hdparm	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/install-mbr		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/jfs_.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/losetup.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -86,6 +88,7 @@
 /usr/sbin/make_reiser4		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/mkdosfs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/mke2fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/mke2fs\.e2fsprogs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/mke4fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/mkfs.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/mkraid		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -94,6 +97,7 @@
 /usr/sbin/mkswap\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/parted		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/partprobe		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/partprobe\.parted		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/partx			--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/raidautorun		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/raidstart		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -107,6 +111,7 @@
 /usr/sbin/swapoff\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/swapon.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/tune2fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/tune2fs\.e2fsprogs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/zdb			--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/zhack			--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/zinject		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
-- 
2.7.4

