From 388603dfcfc156d47d8ed7e60383546e664d17fe Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 20 Feb 2014 03:07:05 -0500
Subject: [PATCH] refpolicy: fix real path for sulogin (util-linux)

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/locallogin.fc |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/locallogin.fc b/policy/modules/system/locallogin.fc
index be6a81b..6c349ed 100644
--- a/policy/modules/system/locallogin.fc
+++ b/policy/modules/system/locallogin.fc
@@ -1,3 +1,5 @@
 
 /sbin/sulogin		--	gen_context(system_u:object_r:sulogin_exec_t,s0)
 /sbin/sushell		--	gen_context(system_u:object_r:sulogin_exec_t,s0)
+
+/usr/sbin/sulogin	--	gen_context(system_u:object_r:sulogin_exec_t,s0)
-- 
1.7.9.5

