From 6ad2d0ae0cd0d44292df62be46e526134c665acb Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Mon, 17 Feb 2014 02:30:30 -0500
Subject: [PATCH] fix real path for brctl

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/admin/brctl.fc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/admin/brctl.fc b/policy/modules/admin/brctl.fc
index ed472f0..2a852b0 100644
--- a/policy/modules/admin/brctl.fc
+++ b/policy/modules/admin/brctl.fc
@@ -1,3 +1,4 @@
 /usr/bin/brctl	--	gen_context(system_u:object_r:brctl_exec_t,s0)
 
 /usr/sbin/brctl	--	gen_context(system_u:object_r:brctl_exec_t,s0)
+/usr/sbin/brctl\.bridge-utils	--	gen_context(system_u:object_r:brctl_exec_t,s0)
-- 
2.7.4

