Subject: [PATCH] refpolicy: fix real path for ntp

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/contrib/ntp.fc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/contrib/ntp.fc b/policy/modules/contrib/ntp.fc
index 830ab32..738b9e6 100644
--- a/policy/modules/contrib/ntp.fc
+++ b/policy/modules/contrib/ntp.fc
@@ -18,6 +18,8 @@
 /usr/sbin/ntpdate	--	gen_context(system_u:object_r:ntpdate_exec_t,s0)
 /usr/sbin/sntp		--	gen_context(system_u:object_r:ntpdate_exec_t,s0)
 
+/usr/bin/ntpd   --      gen_context(system_u:object_r:ntpd_exec_t,s0)
+
 /var/lib/ntp(/.*)?		gen_context(system_u:object_r:ntp_drift_t,s0)
 /var/lib/sntp-kod(/.*)?		gen_context(system_u:object_r:ntp_drift_t,s0)
 /var/db/ntp-kod		--	gen_context(system_u:object_r:ntp_drift_t,s0)
-- 
1.9.1

