From a8a1ca90d19db2ae1af25fe72629c1520ddb442b Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 22 Apr 2014 21:34:57 -0400
Subject: [PATCH] sysadm: allow sysamd exec kadmind

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/roles/sysadm.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/roles/sysadm.te b/policy/modules/roles/sysadm.te
index 30c0c83..3a4a31a 100644
--- a/policy/modules/roles/sysadm.te
+++ b/policy/modules/roles/sysadm.te
@@ -1264,6 +1264,10 @@ optional_policy(`
 	zebra_admin(sysadm_t, sysadm_r)
 ')
 
+optional_policy(`
+	kerberos_exec_kadmind(sysadm_t)
+')
+
 ifndef(`distro_redhat',`
 	optional_policy(`
 		auth_role(sysadm_r, sysadm_t)
-- 
2.7.4

