From 87bab239714a1c38d2cc3ca5645c9cb15c8701e0 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Wed, 19 Feb 2014 02:36:30 -0500
Subject: [PATCH] associate devpts_t (pts) to device_t (devtmpfs) file systems

Upstream-Status: pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/kernel/filesystem.te |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/kernel/filesystem.te b/policy/modules/kernel/filesystem.te
index 0974902..27a47b2 100644
--- a/policy/modules/kernel/filesystem.te
+++ b/policy/modules/kernel/filesystem.te
@@ -179,6 +179,9 @@ files_type(tmpfs_t)
 files_mountpoint(tmpfs_t)
 files_poly_parent(tmpfs_t)
 
+# associate devpts_t (pts) to device_t (devtmpfs)
+dev_associate(devpts_t)
+
 # Use a transition SID based on the allocating task SID and the
 # filesystem SID to label inodes in the following filesystem types,
 # and label the filesystem itself with the specified context.
-- 
1.7.9.5

