Subject: [PATCH] build redhad modules

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Joe Slater <jslater@windriver.com>
Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 configure.ac        | 2 ++
 modules/Makefile.am | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 5eade7c..cc8137f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -639,6 +639,8 @@ AC_CONFIG_FILES([Makefile libpam/Makefile libpamc/Makefile libpamc/test/Makefile
 	po/Makefile.in \
 	Make.xml.rules \
 	modules/Makefile \
+	modules/pam_chroot/Makefile modules/pam_console/Makefile \
+	modules/pam_postgresok/Makefile \
 	modules/pam_access/Makefile \
         modules/pam_debug/Makefile modules/pam_deny/Makefile \
 	modules/pam_echo/Makefile modules/pam_env/Makefile \
diff --git a/modules/Makefile.am b/modules/Makefile.am
index 8da4641..215dd48 100644
--- a/modules/Makefile.am
+++ b/modules/Makefile.am
@@ -47,6 +47,9 @@ SUBDIRS := \
 	pam_debug \
 	pam_deny \
 	pam_echo \
+	pam_chroot \
+	pam_console \
+	pam_postgresok \
 	pam_env \
 	pam_exec \
 	pam_faildelay \
-- 
2.29.2

