diff -urNp saftest.orig/HPI-B.01.01/find-test saftest/HPI-B.01.01/find-test
--- saftest.orig/HPI-B.01.01/find-test	2005-01-22 06:02:26.000000000 +0800
+++ saftest/HPI-B.01.01/find-test	2008-07-07 15:31:05.000000000 +0800
@@ -47,7 +47,7 @@ fi
 
 WHERE="$1"
 
-find "$WHERE" -type f \
+find "$WHERE" -type f -follow\
     \( \
        \( -name "*.c"  \) \
        ! -name \*.o ! -name \*~  \
diff -urNp saftest.orig/HPI-B.01.01/LDFLAGS saftest/HPI-B.01.01/LDFLAGS
--- saftest.orig/HPI-B.01.01/LDFLAGS	2005-12-30 10:24:28.000000000 +0800
+++ saftest/HPI-B.01.01/LDFLAGS	2008-07-07 15:25:58.000000000 +0800
@@ -1,4 +1,4 @@
 # Place your libary file here. 
 # Format: -l<libary name>
 # Example: -l<lopenhpi>
-# -lopenhpi
+ -lopenhpi
diff -urNp saftest.orig/HPI-B.01.01/Makefile saftest/HPI-B.01.01/Makefile
--- saftest.orig/HPI-B.01.01/Makefile	2005-12-30 10:23:11.000000000 +0800
+++ saftest/HPI-B.01.01/Makefile	2008-07-08 10:47:06.000000000 +0800
@@ -17,8 +17,6 @@
 # Authors:
 #       Mi,Jun  <jun.mi@intel.com>
 
-TIMEOUT_CMD = ../t0
-
 top_builddir = .
 
 LOGFILE = $(top_builddir)/log/build_log
@@ -32,7 +30,7 @@ VERBOSE = 0
 
 all: build-tests
 
-build-tests: clean $(TIMEOUT_CMD) $(BUILD_TESTS:.c=.test)
+build-tests: clean $(BUILD_TESTS:.c=.test)
 
 tests-pretty:
 	$(MAKE) all | column -t -s:
@@ -43,7 +41,6 @@ INCLUDE = -Iinclude -I../include 
 
 
 clean:
-	@rm -f $(TIMEOUT_CMD)
 	@rm -f $(LOGFILE)
 	@find $(top_builddir) -iname \*.test | xargs -n 40 rm -f {}
 	@find $(top_builddir) -iname \*~ -o -iname \*.o | xargs -n 40 rm -f {}
@@ -52,7 +49,7 @@ clean:
 %.test: %.o
 	@COMPLOG=$(LOGFILE).$$$$; \
 	[ -f $< ] || exit 0; \
-	nm -g --defined-only $< | grep -q " T main" || exit 0; \
+	$(NM) -g --defined-only $< | grep -q " T main" || exit 0; \
 	if $(CC) $(CFLAGS) $< -o $@ $(LDFLAGS) > $$COMPLOG 2>&1; \
 	then \
 		echo "$(@:.test=): link: PASS" | tee -a $(LOGFILE); \
@@ -88,8 +85,3 @@ clean:
 	fi; \
 	rm -f $$COMPLOG;
 
-$(TIMEOUT_CMD):$(TIMEOUT_CMD).c
-	@echo Building timeout helper files; \
-	$(CC) -O2 -o $@ $< ; \
-
-
