Upstream-Status: Inappropriate [embedded specific]

Pass option '--catalogs' to xsltproc via a2x.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/doc/Makefile.am b/doc/Makefile.am
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 873575f..c63f5fa 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -48,6 +48,6 @@ XSLTPROC_FO_OPTIONS 	?= $(XSLTPROC_OPTIONS)
 	$(MANPAGES_STYLESHEET) $<
 
 hb_report.8: hb_report.8.txt
-	a2x -L -f manpage $<
+	a2x --xsltproc-opts=--catalogs -L -f manpage $<
 
 endif
