Remove bashisms from init script.

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/init/generic.in b/init/generic.in
index 1ce26f0..708e585 100755
--- a/init/generic.in
+++ b/init/generic.in
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # Authors:
 #  Andrew Beekhof <abeekhof@redhat.com>
@@ -28,12 +28,12 @@ PATH="/sbin:/bin:/usr/sbin:/usr/bin:@SBINDIR@"
 
 success()
 {
-	echo -ne "[  OK  ]\r"
+	echo "[  OK  ]"
 }
 
 failure()
 {
-	echo -ne "[FAILED]\r"
+	echo "[FAILED]"
 }
 
 status()
@@ -74,7 +74,7 @@ cluster_disabled_at_boot()
 {
        if grep -q nocluster /proc/cmdline && \
           [ "$(tty)" = "/dev/console" ]; then
-               echo -e "not configured to run at boot"
+               echo "not configured to run at boot"
                failure
                return 1
        fi
@@ -116,7 +116,7 @@ executed_by_cman()
 	[ -f @LOCALSTATEDIR@/run/cman.pid ] || return 0
 
 	read cman_pid foo < @LOCALSTATEDIR@/run/cman.pid
-	if [ "$(pidof $prog)" == "$cman_pid" ];then
+	if [ "$(pidof $prog)" = "$cman_pid" ];then
 		echo -n "$desc was executed by cman"
 		failure
 		echo
diff --git a/init/notifyd.in b/init/notifyd.in
index 9ed7d2d..3809566 100755
--- a/init/notifyd.in
+++ b/init/notifyd.in
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # Authors:
 #  Angus Salkeld <asalkeld@redhat.com>
@@ -27,12 +27,12 @@ PATH="/sbin:/bin:/usr/sbin:/usr/bin:@SBINDIR@"
 
 success()
 {
-	echo -ne "[  OK  ]\r"
+	echo "[  OK  ]"
 }
 
 failure()
 {
-	echo -ne "[FAILED]\r"
+	echo "[FAILED]"
 }
 
 status()
