Remove bashisms from init script.

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/init/corosync-notifyd.in b/init/corosync-notifyd.in
index 3711821..a7cd64a 100755
--- a/init/corosync-notifyd.in
+++ b/init/corosync-notifyd.in
@@ -27,12 +27,12 @@ PATH="/sbin:/bin:/usr/sbin:/usr/bin:@SBINDIR@"
 
 success()
 {
-	echo -ne "[  OK  ]\r"
+	echo "[  OK  ]"
 }
 
 failure()
 {
-	echo -ne "[FAILED]\r"
+	echo "[FAILED]"
 }
 
 status()
diff --git a/init/corosync.in b/init/corosync.in
index c93f971..20044dd 100755
--- a/init/corosync.in
+++ b/init/corosync.in
@@ -28,12 +28,12 @@ PATH="/sbin:/bin:/usr/sbin:/usr/bin:@SBINDIR@"
 
 success()
 {
-	echo -ne "[  OK  ]\r"
+	echo "[  OK  ]"
 }
 
 failure()
 {
-	echo -ne "[FAILED]\r"
+	echo "[FAILED]"
 }
 
 status()
@@ -74,7 +74,7 @@ cluster_disabled_at_boot()
 {
        if grep -q nocluster /proc/cmdline && \
           [ "$(tty)" = "/dev/console" ]; then
-               echo -e "not configured to run at boot"
+               echo "not configured to run at boot"
                failure
                return 1
        fi
@@ -91,7 +91,7 @@ wait_for_ipc()
 		if corosync-cfgtool -s > /dev/null 2>&1; then
 			return 0
 		fi
-		sleep 0.5
+		sleep 1
 		try=$((try + 1))
 	done
 
