From 1bca552e820fe9691866fff83b8b8b59ec37e053 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 25 Apr 2019 11:12:34 +0800
Subject: [PATCH] refpolicy: fix real path for ntp

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/ntp.fc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/services/ntp.fc b/policy/modules/services/ntp.fc
index cd69ea5..49ffe6f 100644
--- a/policy/modules/services/ntp.fc
+++ b/policy/modules/services/ntp.fc
@@ -25,6 +25,7 @@
 /usr/lib/systemd/systemd-timesyncd	--	gen_context(system_u:object_r:ntpd_exec_t,s0)
 
 /usr/sbin/ntpd				--	gen_context(system_u:object_r:ntpd_exec_t,s0)
+/usr/sbin/ntpd\.ntp				--	gen_context(system_u:object_r:ntpd_exec_t,s0)
 /usr/sbin/ntpdate			--	gen_context(system_u:object_r:ntpdate_exec_t,s0)
 /usr/sbin/sntp				--	gen_context(system_u:object_r:ntpdate_exec_t,s0)
 
-- 
2.7.4

