From ea4323538e0da01653cddb78a0abe2aec233d27c Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Mon, 17 Feb 2014 02:22:09 -0500
Subject: [PATCH 1/6] fix real path for arping.iputils

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/admin/netutils.fc |    1 +
 1 file changed, 1 insertion(+)

--- refpolicy.orig/policy/modules/admin/netutils.fc
+++ refpolicy/policy/modules/admin/netutils.fc
@@ -4,6 +4,7 @@
 
 /sbin/arping		--	gen_context(system_u:object_r:netutils_exec_t,s0)
 /bin/arping		--	gen_context(system_u:object_r:netutils_exec_t,s0)
+/usr/sbin/arping\.iputils	--	gen_context(system_u:object_r:netutils_exec_t,s0)
 
 /usr/bin/arping		--	gen_context(system_u:object_r:netutils_exec_t,s0)
 /usr/bin/lft		--	gen_context(system_u:object_r:traceroute_exec_t,s0)
