From 0bfe2282c7a4c36624116bea6f365294f03fb679 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 14 May 2019 09:01:38 +0800
Subject: [PATCH] systemd: fix avc denials for systemd_hw_t

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index c2912cf..fb2719b 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -399,6 +399,8 @@ init_read_state(systemd_hw_t)
 seutil_read_config(systemd_hw_t)
 seutil_read_file_contexts(systemd_hw_t)
 
+corecmd_search_bin(systemd_hw_t)
+
 #######################################
 #
 # locale local policy
-- 
2.7.4

