From e6e2f93914f2812ac558e248564b72452adc371f Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 5 Feb 2016 04:53:09 -0500
Subject: [PATCH] wr-policy: allow auditd use all levels fd

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/logging.te | 1 +
 1 file changed, 1 insertion(+)

--- refpolicy.orig/policy/modules/system/logging.te
+++ refpolicy/policy/modules/system/logging.te
@@ -203,6 +203,7 @@ miscfiles_read_localization(auditd_t)
 
 mls_file_read_all_levels(auditd_t)
 mls_file_write_all_levels(auditd_t) # Need to be able to write to /var/run/ directory
+mls_fd_use_all_levels(auditd_t) # WRL fix - ported from CGP7
 
 seutil_dontaudit_read_config(auditd_t)
 
