--- a/configure.ac
+++ b/configure.ac
@@ -1427,9 +1427,9 @@ dnl ====================================
 AC_CHECK_HEADERS(gnutls/gnutls.h)
 AC_CHECK_HEADERS(security/pam_appl.h pam/pam_appl.h)
 
-dnl GNUTLS library: Attempt to determine by 'libgnutls-config' program.
-dnl If no 'libgnutls-config', try traditional autoconf means.
-AC_PATH_PROGS(LIBGNUTLS_CONFIG, libgnutls-config)
+dnl libgnutls-config has been deprecated, so we use pkg-config
+
+LIBGNUTLS_CONFIG="pkg-config gnutls"
 
 if test -n "$LIBGNUTLS_CONFIG"; then
 	AC_MSG_CHECKING(for gnutls header flags)
