Upstream-Status: Pending

Subject: ash: fix memleak

Signed-off-by: Hu <yadi.hu@windriver.com>
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 shell/ash.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/shell/ash.c b/shell/ash.c
index 7a11305..c17c797 100644
--- a/shell/ash.c
+++ b/shell/ash.c
@@ -8709,6 +8709,7 @@ evaltree(union node *n, int flags)
 	int checkexit = 0;
 	int (*evalfn)(union node *, int);
 	int status = 0;
+	struct stackmark smark;
 
 	if (n == NULL) {
 		TRACE(("evaltree(NULL) called\n"));
@@ -8729,6 +8730,7 @@ evaltree(union node *n, int flags)
 		status = !evaltree(n->nnot.com, EV_TESTED);
 		goto setstatus;
 	case NREDIR:
+		setstackmark(&smark);
 		expredir(n->nredir.redirect);
 		status = redirectsafe(n->nredir.redirect, REDIR_PUSH);
 		if (!status) {
@@ -8736,6 +8738,7 @@ evaltree(union node *n, int flags)
 		}
 		if (n->nredir.redirect)
 			popredir(/*drop:*/ 0, /*restore:*/ 0 /* not sure */);
+		popstackmark(&smark);
 		goto setstatus;
 	case NCMD:
 		evalfn = evalcommand;
-- 
2.12.3

