diff --git a/debian/init.d b/debian/init.d
index 363ba89..330cb2a 100644
--- a/debian/init.d
+++ b/debian/init.d
@@ -10,6 +10,9 @@
 # Description:       Start ippool daemon
 ### END INIT INFO
 
+# Source function library.
+. /etc/init.d/functions
+
 DAEMON=/usr/sbin/ippoold
 NAME=ippoold
 MODULE=pppol2tp
@@ -35,6 +38,10 @@ case "$1" in
     fi
     log_end_msg $?
     ;;
+  status)
+    status /usr/sbin/pppd;
+    exit $?
+    ;;
   restart)
     $0 stop
     sleep 1
@@ -46,7 +53,7 @@ case "$1" in
     log_end_msg $?
     ;;
   *)
-    log_success_msg "Usage: /etc/init.d/ippoold {start|stop|restart|reload|force-reload}"
+    log_success_msg "Usage: /etc/init.d/ippoold {start|stop|status|restart|reload|force-reload}"
     exit 1
 esac
 
