---
 term-utils/script.c |   15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

--- a/term-utils/script.c
+++ b/term-utils/script.c
@@ -109,6 +109,7 @@ int	forceflg = 0;
 
 int die;
 int resized;
+int childstatus;
 
 static void
 die_if_link(char *fn) {
@@ -318,11 +319,11 @@ finish(int dummy __attribute__ ((__unuse
 	int status;
 	pid_t pid;
 
-	while ((pid = wait3(&status, WNOHANG, 0)) > 0)
-		if (pid == child) {
-			childstatus = status;
+	while ((pid = waitpid(-1, &status, WNOHANG)) > 0) {
+		if (pid == child)
 			die = 1;
-		}
+		childstatus = status;
+	}
 }
 
 void
@@ -494,7 +495,11 @@ done(void) {
 		else
 			exit(WEXITSTATUS(childstatus));
 	}
-	exit(EXIT_SUCCESS);
+
+	if (childstatus < 256 && childstatus > 0)
+		exit(childstatus);
+
+	exit(childstatus>>8);
 }
 
 void
