---
 term-utils/script.c |   16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

--- a/term-utils/script.c
+++ b/term-utils/script.c
@@ -618,8 +618,12 @@ doinput(void) {
 	while (die == 0) {
 		memcpy(&rds, &master_rds, sizeof(master_rds));
 		select_ret = select(1, &rds, NULL, NULL, NULL);
-		if (select_ret < 0)
-			break;
+		if (select_ret < 0) {
+			if (errno == EINTR)
+				continue;
+			else
+				break;
+		}
 		if ((cc = read(STDIN_FILENO, ibuf, BUFSIZ)) > 0) {
 			ssize_t wrt = write(master, ibuf, cc);
 			if (wrt < 0) {
@@ -645,8 +649,12 @@ socket_con:
 	while (die == 0) {
 		memcpy(&rds, &master_rds, sizeof(master_rds));
 		select_ret = select(nsockhandle, &rds, NULL, NULL, NULL);
-		if (select_ret < 0)
-			break;
+		if (select_ret < 0) {
+			if (errno == EINTR)
+				continue;
+			else
+				break;
+		}
 		if (FD_ISSET(0, &rds))
 			if (handle_stdin(ibuf) < 0)
 				break;
